src/version.c

```diff
@@ -41,10 +41,16 @@
  * the global version_url is used and the cached version is ignored. */
 int get_latest_version(char *v_url)
 {
+#define MAX_VERSION_CHARS 10
+#define MAX_VERSION_STR_SIZE 11
+
 	char *url = NULL;
 	char *path = NULL;
 	int ret = 0;
-	struct version_container tmp_version = { 0 };
+	struct curl_file_data tmp_version = {
+		MAX_VERSION_STR_SIZE, 0,
+		alloca(MAX_VERSION_STR_SIZE)
```

Comment:
If we know the size then there is no need to use alloca, just do a normal stack allocation.

Reply:
good point. I prefer to avoid alloca too, I just haven't thought about that when I was writing this